test/infrastructure/docker/internal/docker/machine.go

```diff
@@ -212,40 +208,54 @@ func (m *Machine) Create(ctx context.Context, image string, role string, version
 	if m.container == nil {
 		var err error
 
-		machineImage := m.machineImage(version)
+		// Get the KindMapping for the target K8s version.
+		// NOTE: The KindMapping allows to select the most recent kindest/node image available, if any, as well as
+		// provide info about the mode to be used when starting the kindest/node image itself.
+		if version == nil {
+			return errors.New("cannot create a DockerMachine for a nil version")
+		}
+
+		semVer, err := semver.Parse(strings.TrimPrefix(*version, "v"))
+		if err != nil {
+			return errors.Wrap(err, "failed to parse DockerMachine version")
+		}
+
+		kindMapping := kind.GetMapping(semVer)
+
+		// Use custom image if provided.
 		if image != "" {
```

Comment:
Alternative here would be to always use the kind mapping when the image is in the form `kindest/node:vX.X.X`. I'm in favor of this as the current version means adding the same tag in the custom image has different behaviour than a machine that appears to be using that image. I think this could also simplify the logic in nodepool - but I haven't looked closely at that code yet.

Reply:
Done!